docs/guide/using-hooks.md

```diff
@@ -0,0 +1,54 @@
+# Using hooks
+
+dbt provides the ability to run arbitrary commands against the database before and after a model is run. These are known as pre- and post-model hooks and configured as such:
+
+```YAML
+models:
+  project-name:
+    pre-hook:       # arbitrary SQL
+    post-hook:      # arbitrary SQL
+
+```
+
+Hooks are extremely powerful, allowing model authors to perform tasks such as inserting records into audit tables, executing `GRANT` statements, and running `VACUUM` commands, among others.
+
+`pre-hook` and `post-hook` configuration options apply, like all configuration options, to the scope in which they are defined in the model hierarchy. The example above defines them at the level of the entire project; they can also be specified within a given directory of that project or for an individual model.
+
+Multiple instances of `pre-hook` and `post-hook` may be defined. In this case, dbt will run each pre-hook and post-hook (in some order to be determined revisit this).
```

Comment:
The order these get compiled is:

1. in-model dependency hook (if it's a dependency)
2. in-project dependency hook (if it's a dependency)
3. in-model top-level project hook (if it's a top-level model)
4. in-project top-level hook (always, if provided)

1 and 3 are mutually exclusive. A model can either be a dependency model or a top-level model, but of course it can't be both! If a model is a dependency, then 1, 2, and 4 apply. If it's top-level, then only 3 and 4 apply. 

This is a little tricky to explain, so lmk if it's not clear